test/unit/event_store/event_test.rb

```diff
@@ -86,4 +86,42 @@ def test_stale
     expected_stale_events_count = EventStore::Event.where('created_at <= ?', EventStore::Event::TTL.ago).count
     assert_equal expected_stale_events_count, EventStore::Event.stale.count
   end
+
+  class EventRollbacksTest < ActiveSupport::TestCase
+
+    class EventWithCallbacks < EventStore::Event
+      after_rollback :after_rollback_method
+
+      def after_rollback_method; end
+    end
+
+    test 'rolledback harmless exceptions not reported' do
+      System::ErrorReporting.expects(:report_error).with(instance_of(EventStore::Event::EventRollbackError)).never
+
+      ActiveRecord::Base.transaction do
+        EventStore::Event.create!(stream: 'dummy', event_type: 'Dummy', event_id: 1, provider_id: 1, data: { whatever: 1 })
+        ActiveJob::Arguments.expects(:deserialize).raises(ActiveRecord::RecordNotFound).at_least_once
+        raise ActiveRecord::Rollback
+      end
+    end
+
+    test 'failed rolledback with transactional callbacks' do
```

Comment:
```suggestion
    test 'failed rollback with transactional callbacks are reported' do
```

Reply:
`rolledback` is the method name, so it makes sense to stay. It's not just any random rollback error.

Reply:
How about these @akostadinov @jlledom ?

https://github.com/3scale/porta/pull/3705/commits/7e1c0146650e24521cf00d569ea0c04711766a01

Reply:
Works for me.